kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,872 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "snake_sbm_utilities.h"
+
+namespace Kratos
+{
+
+    void SnakeSbmUtilities::CreateTheSnakeCoordinates(
+        ModelPart& rIgaModelPart,
+        ModelPart& rSkinModelPartInnerInitial,
+        ModelPart& rSkinModelPartOuterInitial, 
+        ModelPart& rSkinModelPart, 
+        int rEchoLevel, 
+        Vector& knotVectorU, 
+        Vector& knotVectorV,
+        const std::size_t numberOfInnerLoops,
+        const double lambdaInner,
+        const double lambdaOuter)
+    {   
+
+        // Initilize the property of skin_model_part_in and out
+        if (rSkinModelPartInnerInitial.NumberOfNodes()>0) {
+            // template argument IsInnerLoop set true
+            CreateTheSnakeCoordinates<true>(rIgaModelPart, rSkinModelPartInnerInitial, rSkinModelPart, knotVectorU, knotVectorV, numberOfInnerLoops, lambdaInner);
+                
+        }
+        if (rSkinModelPartOuterInitial.NumberOfNodes()>0) {
+            // template argument IsInnerLoop set false
+            CreateTheSnakeCoordinates<false>(rIgaModelPart, rSkinModelPartOuterInitial, rSkinModelPart, knotVectorU, knotVectorV, 1, lambdaOuter);
+        }
+    }   
+
+    template <bool TIsInnerLoop>
+    void SnakeSbmUtilities::CreateTheSnakeCoordinates(
+        ModelPart& rIgaModelPart, 
+        ModelPart& rSkinModelPartInitial,
+        ModelPart& rSkinModelPart, 
+        Vector& knotVectorU, 
+        Vector& knotVectorV,
+        const std::size_t numberOfLoops,
+        const double lambda) 
+    { 
+        int r_echo_level = GetEchoLevel();
+        const bool is_inner = TIsInnerLoop;
```

Comment:
Why you need this variable? Indeed, use the template argument and `if constexpr` clauses.